File: backend/src/controllers/schedulerController.js
const schedulerService = require('../services/schedulerService');

class SchedulerController {
  // 启用定时任务
  async enableSchedule(req, res) {
    try {
      const { cronExpression } = req.body;
      const config = await schedulerService.enableSchedule(cronExpression);
      
      res.json({
        success: true,
        data: config,
        message: '定时任务已启用'
      });
    } catch (error) {
      console.error('启用定时任务错误:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  // 禁用定时任务
  async disableSchedule(req, res) {
    try {
      const config = schedulerService.disableSchedule();
      
      res.json({
        success: true,
        data: config,
        message: '定时任务已禁用'
      });
    } catch (error) {
      console.error('禁用定时任务错误:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  // 获取定时任务配置
  async getConfig(req, res) {
    try {
      const config = schedulerService.getConfig();
      
      res.json({
        success: true,
        data: config,
        message: '获取成功'
      });
    } catch (error) {
      console.error('获取配置错误:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  // 手动触发一次
  async triggerNow(req, res) {
    try {
      const result = await schedulerService.triggerNow();
      
      res.json({
        success: true,
        data: result,
        message: '爬虫任务已触发'
      });
    } catch (error) {
      console.error('触发任务错误:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  // 获取预设的cron表达式
  async getPresets(req, res) {
    try {
      const presets = schedulerService.getPresetCronExpressions();
      
      res.json({
        success: true,
        data: presets,
        message: '获取成功'
      });
    } catch (error) {
      console.error('获取预设错误:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }
}

module.exports = new SchedulerController();

File: creator-data-management/src/main/java/com/data/creator/api/SchedulerController.java
package com.data.creator.api;

import com.data.creator.dto.ApiResponse;
import com.data.creator.services.SchedulerService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@Slf4j
@RestController
@RequestMapping("/api/scheduler")
@RequiredArgsConstructor
public class SchedulerController {

    private final SchedulerService schedulerService;

    @GetMapping("/config")
    public ApiResponse<Map<String, Object>> config() {
        return ApiResponse.ok(schedulerService.config());
    }

    @GetMapping("/presets")
    public ApiResponse<List<Map<String, String>>> presets() {
        return ApiResponse.ok(schedulerService.presets());
    }

    @PostMapping("/enable")
    public ApiResponse<Map<String, Object>> enable(@RequestBody Map<String, String> req) {
        String cron = req.get("cronExpression");
        if (cron == null || cron.isEmpty()) {
            return ApiResponse.fail("cronExpression required");
        }
        try {
            return ApiResponse.ok(schedulerService.enable(cron));
        } catch (Exception e) {
            return ApiResponse.fail("enable failed: " + e.getMessage());
        }
    }

    @PostMapping("/disable")
    public ApiResponse<Map<String, Object>> disable() {
        return ApiResponse.ok(schedulerService.disable());
    }

    @PostMapping("/trigger")
    public ApiResponse<Map<String, Object>> trigger() {
        schedulerService.trigger();
        return ApiResponse.ok(Map.of("message", "triggered"));
    }
}
